src/vcf/assembly_checker.cpp

```diff
@@ -22,10 +22,14 @@ namespace ebi
   {
     namespace assembly_checker
     {
+      bool process_vcf_records(std::istream & vcf_input,
+                               std::shared_ptr<ebi::vcf::fasta::IFasta> & fasta,
+                               const std::string & assembly_report,
+                               std::vector<std::unique_ptr<ebi::vcf::AssemblyCheckReportWriter>> & outputs);
+
```

Comment:
I think we can either shift this declaration to hpp file or we can move `process_vcf_ref` declaration here. I would go with moving `process_vcf_ref` to this file. we may keep only those declarations in hpp which are used elsewhere in other files. what do you say @cyenyxe @jmmut

Reply:
I would like leaving only one function in the `assembly_checker` header with current implementation and forward declare other functions. I didn't do it because I did not want to mix this kind of refactoring with a feature development.